hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1047,50 +1047,158 @@ public function createInsurancePlanTypes() {
   
   /**
    * Upgrader to :
-   * 1- Remove all pay scales except 'Not Applicable'
-   * 2- Remove all Hour Locations except 'Head Office'
-   * 3- Add 'Fixed Term' Contract Type
-   * 4- Sort contract types alphabetically
-   * 5- Add 'Retirement' contract end reason
-   * 6- add new contract change reason options
+   * 
+   * - Remove unused pay scales except 'Not Applicable'
+   * - Remove unused Hour Locations except 'Head Office'
+   * - Remove Duplicated 'Employee - Permanent' Contract Type
+   * - Add 'Fixed Term' Contract Type
+   * - Sort contract types alphabetically
+   * - Add 'Retirement' contract end reason
+   * - Add new contract change reason options
    *
    * @return TRUE
    */
   public function upgrade_1026() {
     $this->up1026_removeUnneededPayScales();
     $this->up1026_removeUnneededHourLocations();
+    $this->up1026_removeDuplicateContractType();
+    $this->up1026_removeRevisionChangeReasons();
 
     $optionValues = [
-      ['hrjc_contract_type', ['Fixed Term']],
-      ['hrjc_contract_end_reason', ['Retirement']],
-      ['hrjc_revision_change_reason', ['Promotion', 'Increment', 'Disciplinary'] ],
+      'hrjc_contract_type' => ['Fixed Term'],
+      'hrjc_contract_end_reason' => ['Retirement'],
+      'hrjc_revision_change_reason' => ['Promotion', 'Increment', 'Disciplinary'],
     ];
 
-    foreach ($optionValues as $value) {
-      $this->addOptionValues($value[0], $value[1]);
+    foreach ($optionValues as $optionGroup => $values) {
+      $this->addOptionValues($optionGroup, $values);
     }
 
     $this->up1026_sortContractTypes();
 
     return true;
   }
 
+  private function up1026_removeRevisionChangeReasons() {
+    $deleteableReasons = [];
+    $reasons = $this->getEntityRecords('OptionValue', ['option_group_id' => 'hrjc_revision_change_reason']);
+
+    foreach ($reasons as $currentReason) {
+      $q = "
+        SELECT id
+        FROM civicrm_hrjobcontract_revision
+        WHERE change_reason = '{$currentReason['value']}'
+      ";
+      $reasonInContracts = CRM_Core_DAO::executeQuery($q);
+
+      if (!$reasonInContracts->fetch()) {
+        $deleteableReasons[] = $currentReason['id'];
+      }
+    }
+
+    $this->deleteEntityRecords('HRHoursLocation', $deleteableReasons);
+  }
+
   /**
-   * Removes all Pay scales except 'Not Applicable'
+   * Removes duplicates for 'Employee - Permanent' contract type.
+   */
+  private function up1026_removeDuplicateContractType() {
```

Comment:
No need to do the check, since what is stored in DB on contract details table is not the ID of the contract type, but the value (ie. literally "Employee - Permanent").